R/zzz.R

```diff
@@ -277,6 +277,18 @@ default_undesirable_operators <- all_undesirable_operators[names(all_undesirable
 default_settings <- NULL
 
 settings <- NULL
+# TODO(R>=3.6.0): Just use sys.source() directly. Note that we can't
```

Comment:
Can't we already bump the minimum needed R version to `3.6.0`?

`{styler}` already [has](https://cran.r-project.org/web/packages/styler/index.html), and this is also not violating tidyverse/r-lib's [stated policy](https://www.tidyverse.org/blog/2019/04/r-version-support/).

<img width="809" alt="Screenshot 2023-09-18 at 22 44 40" src="https://github.com/r-lib/lintr/assets/11330453/5f5ad70a-a724-4305-ab0a-d9e4087312ee">

Reply:
I still don't see any "core" tidyverse packages using that -- I'd wait for them to make the bump first:

https://github.com/tidyverse/dplyr/blob/73742713bc61d4ef079e72b806143b23cb7c4871/DESCRIPTION#L22-L23

Reply:
Note the core packages drop testing for the old versions IINM. They don't explicitly update the `Depends` field.

Reply:
Let's defer to next release, otherwise I might prefer making this release 3.2.0.